src/app/main-page/presentation/toolbar-action/toolbar-action.component.ts

```diff
@@ -128,14 +130,29 @@ export class ToolbarActionComponent implements OnChanges {
         this.graphType.emit(this.graphSettings.type);
     }
 
-    setMapType(mapType: MapType) {
-        this.mapType.emit(mapType);
+    setMapSettings(userInput: MapType | TileServer): void {
+        const inputTypeTile = Object.values(TileServer).includes(
+            TileServer[userInput],
+        );
+
+        if (inputTypeTile) {
+            this.mapSettings.emit({
+                mapType: MapType.TILES_ONLY,
+                tileServer: TileServer[userInput],
+            });
+            return;
+        }
+
+        this.mapSettings.emit({
+            mapType: MapType[userInput],
+            tileServer: this.graphSettings.tileServer,
+        });
     }
 
     onSelectShapeFile(event): void {
         // this is necessary, otherwise the 'Load Shape File...' option might stay active
         setTimeout(() => {
-            this.selectedMapTypeOption = "" + this._graphSettings.mapType;
+            this.selectedOption = "" + MapType.SHAPE_ONLY;
```

Comment:
This should not work. You set the `selectedOption` independently on the success on the ShapeFile Loading ...

You could use `this.updateSelectedMapOption();` instead
with 
```
private updateSelectedMapOption(): void {
    this.selectedMapOption = this._graphSettings.mapType === MapType.TILES_ONLY ?
            this._graphSettings.tileServer : this._graphSettings.mapType;
}
```
then you could use this function on input set to reduce redundant code.

Reply:
- [ ] use func to update selected map option

Reply:
moving this code to the setMapType setTileServer methods will result in selectShapeFileOption staying active when the upload is cancelled. I'll use your snippet, but I'll call it in the timeout as it has been done before.